mathics/core/expression.py

```diff
@@ -743,7 +755,13 @@ def flatten(
         else:
             return self
 
-    def evaluate(self, evaluation) -> typing.Union["Expression", "Symbol"]:
+    def evaluate(self, evaluation) -> "BaseExpression":
```

Comment:
This is wrong. What is returned is some subclass of "Expression". I corrected this now in #134 




Furthermore, this I believe is something to be important.   We start

Reply:
@rocky, ``evaluate`` can return a  ``String`` or a `PrecisionReal` number, not necessarily an  ``Expression``.